src/v2/guide/custom-directive.md

```diff
@@ -89,6 +89,8 @@ Directive hooks are passed these arguments:
 - `vnode`: The virtual node produced by Vue's compiler. See the [VNode API](../api/#VNode-Interface) for full details.
 - `oldVnode`: The previous virtual node, only available in the `update` and `componentUpdated` hooks.
 
+<p class="tip">Directive arguments can be dynamic. For example, in `v-my-directive:[foo]`, the arg would be the value of the data property `foo` on your component instance. As directive hooks are invoked, the value of `arg` within the `binding` argument will dynamically change based on the value of `foo`.</p>
```

Comment:
This is great! I think it would be more clear to use an example like `v-mydirective:arg="[dataproperty]"`

The two differences there being:
- `v-my-directive` might be more confusing for people just learning
- Using more "real" names and a case that would work can be a little clearer.

It also needs to be taken out of a tip, as there's a tip right next to it, and we try not to do two in a row like that. 

I've made a quick example that can be included as well. If you don't feel like including, I can just follow up with another PR after yours.

https://codepen.io/sdras/pen/62a58fc1a893bc88c7a6795d212b6a8c?editors=1010

Great work! Thanks!

Reply:
You're welcome. I made some changes based on your comments.